app_dart/lib/src/request_handlers/refresh_chromebot_status.dart

```diff
@@ -90,8 +90,31 @@ class RefreshChromebotStatus extends ApiRequestHandler<Body> {
           update.builderName = builder.name;
           update.luciBucket = 'luci.flutter.prod';
           await datastore.insert(<Task>[update]);
+          // Save luci task record to BigQuery only when task finishes.
+          if (update.status == Task.statusFailed || update.status == Task.statusSucceeded) {
+            await _insertBigquery(update);
+          }
         }
       }
     }
   }
+
+  Future<void> _insertBigquery(Task task) async {
```

Comment:
should this be a generic function that works for devicelab, luci, etc? if it is let's move to a utilities class and reuse instead if rewrite here.

Reply:
The generic function is `insertBigquery` defined in `utils.dart`. Here the `_insertBigquery` only prepares the data to be injected to the generic function.